Mathlib/Util/AssertExists.lean

```diff
@@ -49,4 +49,7 @@ You should *not* delete the `assert_not_exists` statement without careful discus
 -/
 elab "assert_not_exists " n:ident : command =>
   do let [] ← try resolveGlobalConstWithInfos n catch _ => return
-      | throw <| Exception.error n m!"Declaration {n} is not allowed to exist in this file"
+      | throw <| .error n <| m!"Declaration {n} is not allowed to be imported by this file.\n\n " ++
+          "These invariants are maintained by `assert_not_exists` statements, " ++
+          "and exist in order to ensure that \"complicated\" parts of the library " ++
+          "are not accidentally introduced as dependencies of \"simple\" parts of the library."
```

Comment:
I'm going to commit this in a moment -- I'd like to use the standard hack for multiline strings so that I can search-replace once string gaps land in the next Lean version bump.
```suggestion
      | throw <| .error n <| m!"Declaration {n} is not allowed to be imported by this file.\n\n{
           ""}These invariants are maintained by `assert_not_exists` statements, {
           ""}and exist in order to ensure that \"complicated\" parts of the library {
           ""}are not accidentally introduced as dependencies of \"simple\" parts of the library."
```